src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -636,6 +654,60 @@ private void validateComplaintsConfig(Course course) {
     }
 
+    /**
+     * GET /courses/:courseId/submissions Get all submissions for course
+     *
+     * @param courseId the id of the course
+     * @return the ResponseEntity with status 200 (OK) and with body the course, or with status 404 (Not Found)
+     * @throws AccessForbiddenException if the current user doesn't have the permission to access the course
+     */
+    @GetMapping("/courses/{courseId}/submissions")
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<List<Submission>> getSubmissionsForCourse(@PathVariable Long courseId) throws AccessForbiddenException {
+        log.debug("REST request to get all submissions for course : {}", courseId);
+        long start = System.currentTimeMillis();
+        Course course = courseService.findOneWithExercises(courseId);
+        User user = userService.getUserWithGroupsAndAuthorities();
+        if (!authCheckService.isAtLeastInstructorInCourse(course, user)) {
+            throw new AccessForbiddenException("You are not allowed to access this resource");
+        }
+
+        List<Exercise> exercises = exerciseService.findAllForCourse(course, user);
+        List<Submission> submissions = new ArrayList<>();
+
+        for (Exercise exercise : exercises) {
+            if (exercise instanceof ModelingExercise) {
+                List<ModelingSubmission> modelingSubmissions = modelingSubmissionService.getModelingSubmissions(exercise.getId(), true);
+                for (ModelingSubmission modelingSubmission : modelingSubmissions) {
+                    modelingSubmission.getParticipation().setExercise(exercise);
+                }
+                submissions.addAll(modelingSubmissions);
+            }
+            else if (exercise instanceof TextExercise) {
+                List<TextSubmission> textSubmissions = textSubmissionService.getTextSubmissionsByExerciseId(exercise.getId(), true);
+                for (TextSubmission textSubmission : textSubmissions) {
+                    textSubmission.getParticipation().setExercise(exercise);
+                }
+                submissions.addAll(textSubmissions);
+            }
+            else if (exercise instanceof FileUploadExercise) {
+                List<FileUploadSubmission> fileUploadSubmissions = fileUploadSubmissionService.getFileUploadSubmissions(exercise.getId(), true);
+                for (FileUploadSubmission fileUploadSubmission : fileUploadSubmissions) {
+                    fileUploadSubmission.getParticipation().setExercise(exercise);
+                }
+                submissions.addAll(fileUploadSubmissions);
+            }
+
+            Set<Submission> set = new HashSet<>(submissions);
+            submissions.clear();
+            submissions.addAll(set);
+
+        }
+        long end = System.currentTimeMillis();
+        log.info("Finished /courses/" + courseId + "/submissions call in " + (end - start) + "ms");
+        return ResponseEntity.ok(submissions);
```

Comment:
Make sure to filter (i.e. hide) elements that should not be visible to tutors, e.g. they are not allowed to see the author of the submission

Reply:
Thanks for the hint, done so.